pkg/requestutils/request.go

```diff
@@ -24,14 +27,26 @@ func New(logger lumber.Logger) core.Requests {
 	}
 }
 
-func (r *requests) MakeAPIRequest(ctx context.Context, httpMethod, endpoint string, body []byte) ([]byte, error) {
-	req, err := http.NewRequestWithContext(ctx, httpMethod, endpoint, bytes.NewBuffer(body))
+func (r *requests) MakeAPIRequestWithAuth(ctx context.Context, httpMethod, endpoint string, body []byte) ([]byte, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		r.logger.Errorf("error while parsing endpoint %s, %v", endpoint, err)
+		return nil, err
+	}
+	q := u.Query()
+	q.Set("repoID", os.Getenv("REPO_ID"))
+	q.Set("buildID", os.Getenv("BUILD_ID"))
+	q.Set("orgID", os.Getenv("ORG_ID"))
+	u.RawQuery = q.Encode()
 
+	req, err := http.NewRequestWithContext(ctx, httpMethod, u.String(), bytes.NewBuffer(body))
 	if err != nil {
 		r.logger.Errorf("error while creating http request %v", err)
 		return nil, err
 	}
 
+	req.Header.Add("Authorization", fmt.Sprintf("%s %s", "Bearer", os.Getenv("TOKEN")))
+
```

Comment:
accept headers array as a parameter in this function and pass it as arguments from calling functions. Append all the (passed) headers here. In that way it'll still be a generic function to send API requests. You'll need to refactor the name also to the previous function name then.